### PR TITLE
[13_0_HeavyIon_X] Pythia8 photon flux

### PR DESCRIPTION
**Pull request by stahlleiton**
#### PR description:

backport of https://github.com/cms-sw/cmssw/pull/43840 and https://github.com/cms-sw/cmssw/pull/44264

### CODE REVIEW

**Comment by mandrenguyen** [comment]
backport of https://github.com/cms-sw/cmssw/pull/43840

**Comment by mandrenguyen** [comment]
please test

**Comment by stahlleiton** [comment]
@menglu21 : This backport include also the fixes from https://github.com/cms-sw/cmssw/pull/44264

File: GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc [hunk 1]
```diff
@@ -77,6 +77,43 @@ namespace CLHEP {
...
 
+//Insert class for use w/ PDFPtr for proton-photon flux
+//parameters hardcoded according to main70.cc in PYTHIA8 v3.10
+class Nucleus2gamma2 : public Pythia8::PDF {
+public:
+  // Constructor.
+  Nucleus2gamma2(int idBeamIn) : Pythia8::PDF(idBeamIn) {}
+
+  // Update the photon flux.
+  void xfUpdate(int, double x, double) override {
```
[the remark this answers is not in the document]
Comment (stahlleiton) [reply GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:88]: This change comes from https://github.com/cms-sw/cmssw/pull/44264

File: GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc [hunk 2]
```diff
@@ -77,6 +77,43 @@ namespace CLHEP {
...
+    // lead
+    double radius = 0;  // radius in [fm]
+    double z = 0;
+    if (idBeam == 1000822080) {
```
[the remark this answers is not in the document]
Comment (stahlleiton) [reply GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:92]: This change comes from https://github.com/cms-sw/cmssw/pull/44264

File: GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc [hunk 3]
```diff
@@ -77,6 +77,43 @@ namespace CLHEP {
...
+      radius = 6.636;
+      z = 82;
+    }
+    // oxygen
+    else if (idBeam == 80160) {
```
[the remark this answers is not in the document]
Comment (stahlleiton) [reply GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:97]: This change comes from https://github.com/cms-sw/cmssw/pull/44264

File: GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc [hunk 4]
```diff
@@ -371,6 +414,11 @@ bool Pythia8Hadronizer::initializeForInternalPartons() {
     fMasterGen->settings.word("Beams:LHEF", lheFile_);
   }
 
+  if (doProtonPhotonFlux) {
+    photonFlux = make_shared<Nucleus2gamma2>(1000822080);
```
[the remark this answers is not in the document]
Comment (stahlleiton) [reply GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:418]: This change comes from https://github.com/cms-sw/cmssw/pull/44264

**Comment by menglu21** [comment]
> @menglu21 : This backport includes also the fixes from #44264

I see, thanks for the reminder

**Comment by menglu21** [comment]
+1

**Comment by antoniovilela** [comment]
+1

**Pull request merged**